pkg/pmem-csi-driver/parameters.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2019 Intel Corporation
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pmemcsidriver
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/docker/go-units"
+)
+
+type persistencyModel string
+type parameterOrigin int
+
+// Beware of API and backwards-compatibility breaking when changing these string constants!
+const (
+	parameterCacheSize        = "cacheSize"
+	parameterEraseAfter       = "eraseafter"
+	parameterName             = "name"
+	parameterPersistencyModel = "persistencyModel"
+	parameterVolumeID         = "_id"
+	parameterSize             = "size"
+
+	// Kubernetes v1.16+ adds this key to NodePublishRequest.VolumeContext
+	// while provisioning ephemeral volume.
+	parameterEphemeral = "csi.storage.k8s.io/ephemeral"
+
+	// Additional, unknown parameters that are okay.
+	parameterPodInfoPrefix = "csi.storage.k8s.io/"
+
+	persistencyNormal    persistencyModel = "normal" // In releases <= 0.6.x this was called "none", but not documented.
+	persistencyCache     persistencyModel = "cache"
+	persistencyEphemeral persistencyModel = "ephemeral" // only used internally
+
+	// Parameters are from storage class in controller CreateVolume.
+	createVolumeParameters parameterOrigin = iota
+	// Parameters are from master in node CreateVolume.
+	createVolumeInternalParameters
+	// Parameters are for an ephemeral volume in NodePublishVolume.
+	ephemeralVolumeParameters
+	// Parameters are for a persistent volume in NodePublishVolume.
+	persistentVolumeParameters
+	// Parameters as stored in node volume list.
+	nodeVolumeParameters
+)
+
+// validParameters is a whitelist of which parameters are valid in which context.
+var validParameters = map[parameterOrigin][]string{
+	// Parameters from Kubernetes and users for a persistent volume.
+	createVolumeParameters: []string{
+		parameterCacheSize,
+		parameterEraseAfter,
+		parameterPersistencyModel,
+		parameterPodInfoPrefix,
```

Comment:
Pod info is not valid in this context I guess.

Reply:
This is what we get in `CreateVolume`. We have enabled pod info, so we get those additional parameters.

Reply:
The PodInfo info is passed to `NodePublishVolume` only I guess, not to `CreateVolume`. Please correct me if I am wrong.

Reply:
You are probably right, I was confused about where pod info is provided. I'll fix this.

I still prefer to keep `persistentVolumeParameters` and `createVolumeParameters` separate even if they (currently) have the same content because conceptually they are different.